### PR TITLE
"Peek" launch mode for bluespace launchpads

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
This pr adds a new launch option for bluespace lauchpads called peek. What it will do is launch, wait 3 seconds, then pull you back. Additionaly, hitting peek on multiple pads sequentially will cause them to pull in reverse order, allowing you to utilize this with chained pads.
## Why It's Good For The Game
Launchpads are very hard to use effectively without a partner, as you can teleport yourself into danger with no way out. This will let you more easily trial-error coordinates without removing the risk factor, since you will still be in the area for 5-6 seconds, plenty of time to be spotted, shot at, die in space, etc. 
## Changelog
:cl:
add: bluespace launchpad peek mode
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
>Launchpads are very hard to use effectively without a partner

and what's wrong with that?

**Comment by etherware-novice** [comment]
theres only one person who wants to do it witg me :(

**Comment by Zonespace27** [comment]
Use circuits or find a buddy if you want to do launchpad shenanigans imo

File: code/game/machinery/computer/launchpad_control.dm [hunk 1]
```diff
@@ -183,4 +185,34 @@
...
 			. = TRUE
+
+		if("peek")
+			var/checks = teleport_checks(current_pad)
+			if(!isnull(checks))
+				to_chat(usr, span_warning(checks))
+				return
+
+			current_pad.doteleport(usr, TRUE)
+			if(!LAZYLEN(peek_stack))  // dont pull multiple times
```
Comment (Zonespace27) [inline code/game/machinery/computer/launchpad_control.dm:196]: List isn't lazy

File: code/game/machinery/computer/launchpad_control.dm [hunk 2]
```diff
@@ -183,4 +185,34 @@
...
+
+			. = TRUE
 	. = TRUE
+
+
+/obj/machinery/computer/launchpad/proc/peek_return(user)
+	if(!length(peek_stack))
+		return
+
+	for(var/obj/machinery/launchpad/current_pad in reverseList(peek_stack))
```
Comment (Zonespace27) [inline code/game/machinery/computer/launchpad_control.dm:208]: As anything

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by antropod** [comment]
Just enough time to card AI
but to be honest, just use pair of quantum spin inverters, leave one near console, take other with you
send yourself, do the job, use spin inverter to get back, then pull spin inverter

**Comment by bob-b-b** [comment]
As people above have said, launchpad can be used solo with inverter or by utilizing circuits.

This seems like an unnecessary buff that devalues the inverter and one of few useful aspects of circuits.

**Comment by private-tristan** [comment]
Inverters will always be an amazing getaway option as you can always hide the other one (and break the link to stop others from following), all this change means is that you don't need to spend 5 minutes getting a circuit remote/BCI whenever you want to use launchpads by yourself.

**Comment by Wallemations** [comment]
If the justification for this is "launchpads are difficult to use alone" then I think the PR is pointless. The fact that it takes more than one person to return you from a location is a balancing factor, that makes it so you can't just jump into a locked zone and leave without incident all on your own. 

This is of course overlooking circuits, but it takes time to set up the circuits required for the pad, which also balances it out.

**Comment by Mothblocks** [comment]
Power creep

**Pull request closed without merging**